src/chunk.c

```diff
@@ -41,12 +47,242 @@ chunk_create_from_tuple(HeapTuple tuple, int16 num_constraints)
 	return chunk;
 }
 
+/*
+ * Align a chunk's hypercube in 'aligned' dimensions.
+ *
+ * After a chunk collision scan, this function is called for each chunk in the
+ * chunk scan context. Chunks in the scan context may have only a partial set of
+ * slices if they only overlap in some, but not all, dimensions (see
+ * illustrations below). Still, partial chunks may still be of interest for
+ * alignment in a particular dimension. Thus, if a chunk has an overlapping
+ * slice in an aligned dimension, we cut the context's hypercube to not overlap
+ * with that slice.
+ */
+static bool
```

Comment:
Nit may be worth pointing out the differences between do_dimension_alignment and do_collision resolution. Namely, if I understand correctly, you want to cut aligned dimensions even if the actual hypercube does not collide so that future hypercubes could use the same values in aligned dimensions.

Reply:
Also maybe note that actual alignment (using existing slices) happens elsewhere or change function name to indicate that this is only collision handling

Reply:
Added more elaborate comment, including illustrations.